phpunit/block-supports/background-test.php

```diff
@@ -137,6 +137,23 @@ public function data_background_block_support() {
 				'expected_wrapper'    => '<div style="background-image:url(&#039;https://example.com/image.jpg&#039;);background-size:cover;">Content</div>',
 				'wrapper'             => '<div>Content</div>',
 			),
+			'background image style with contain, position, and repeat is applied' => array(
+				'theme_name'          => 'block-theme-child-with-fluid-typography',
+				'block_name'          => 'test/background-rules-are-output',
+				'background_settings' => array(
+					'backgroundImage' => true,
+				),
+				'background_style'    => array(
+					'backgroundImage'  => array(
+						'url'    => 'https://example.com/image.jpg',
+						'source' => 'file',
+					),
+					'backgroundRepeat' => 'no-repeat',
+					'backgroundSize'   => 'contain',
+				),
+				'expected_wrapper'    => '<div style="background-image:url(&#039;https://example.com/image.jpg&#039;);background-position:center;background-repeat:no-repeat;background-size:contain;">Content</div>',
```

Comment:
> &#039;

Just curious, where is this being escaped again? In the tests or WordPress kses/somewhere?

Reply:
The escaping happens during final output via the HTML tag processor when we call `set_attribute` as it escapes values via `esc_attr` [here](https://github.com/WordPress/wordpress-develop/blob/e9f335df2f990866f8b8f1acbe1852c592dec3e6/src/wp-includes/html-api/class-wp-html-tag-processor.php#L2364). It was discussed a little bit here (https://github.com/WordPress/gutenberg/issues/54336#issuecomment-1782930411), but doesn't appear to cause any issues for browsers. It might be worth us tidying up at some point, but doesn't appear to be particularly urgent since it's only the server-rendered output rather than anything stored anywhere.

Happy to look into any alternatives to workaround the odd escaping if you have any ideas!

Reply:
Thank for the context!